controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -119,6 +123,30 @@ func (w *Workload) getConfigMap(ctx context.Context, configMap ctrlclient.Object
 	return original.DeepCopy(), nil
 }
 
+var firstUnsupportedKubernetesManagementClusterVersion = semver.MustParse("1.22.0")
+
+// IsKubernetesVersionSupported checks if the Kubernetes version is supported.
+// Kubernetes versions >= v1.22.0 for management clusters are not supported.
+// Management clusters are identified by the existence of the KubeadmControlPlane CRD.
+func (w *Workload) IsKubernetesVersionSupported(version semver.Version) (bool, error) {
+	// Kubernetes version is < v1.22.0.
+	if version.LT(firstUnsupportedKubernetesManagementClusterVersion) {
+		return true, nil
+	}
+
+	// Try to get *metav1.PartialObjectMetadata of KubeadmControlPlane CRD.
+	if _, err := w.MetadataClient.Resource(apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions")).Get("kubeadmcontrolplanes.controlplane.cluster.x-k8s.io", metav1.GetOptions{}); err != nil {
```

Comment:
Can we split this line into multiple ones? It's hard to read or review

Reply:
Will be solved by dropping the metadata client